src/JJMasterData.Core/DataDictionary/Repository/DataDictionaryIO.cs

```diff
@@ -0,0 +1,223 @@
+﻿using JJMasterData.Commons.Dao;
+using JJMasterData.Commons.Dao.Entity;
+using JJMasterData.Commons.DI;
+using JJMasterData.Commons.Language;
+using Newtonsoft.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Linq;
+
+namespace JJMasterData.Core.DataDictionary.Repository;
+
+/// <summary>
+/// The Data Dictionaries (metadata) are stored in files in a custom folder
+/// </summary>
+public class DataDictionaryIO : IDataDictionaryRepository
+{
+    private readonly IEntityRepository _entityRepository;
+
+    public string FolderPath { get; }
+    
+    public DataDictionaryIO(IEntityRepository entityRepository)
+    {
+        _entityRepository = entityRepository;
+        FolderPath = "DataDictionaries";
```

Comment:
FolderPath should be stored at `JJMasterDataOptions` (`DataDictionaryIO` is not a plugin), so the user can set the path at `IConfiguration`. In my personal opnion, I think initializating a option by property is hard to understand and by DI it will be hard to set outside the constructor.